vulnfeeds/cpp/main.go

```diff
@@ -0,0 +1,238 @@
+		if err != nil {
+			panic(err)
+		}
+		err = savannahGitRepoTemplate.Execute(&tpl, domain)
+		return tpl.String(), true
+	}
+
+	// Return the original URL unmodified
+	return u, false
+}
+
+// Returns the data associated with the ^Source: line of a machine-readable Debian copyright file
+// See https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+func MaybeGetSourceFromDebianCopyright(copyrightFile string) (string, bool) {
+	re := regexp.MustCompile(`^Source: (.*)$`)
+
+	file, err := os.Open(copyrightFile)
+	if err != nil {
+		Logger.Fatalf("%v", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	// Check the first line to see if we have a machine-readable copyright file
+	scanner.Scan()
+	if scanner.Text() != "Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/" {
+		return "", false
+	}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		match := re.FindStringSubmatch(line)
+		if match != nil {
+			return match[1], true
+		}
+	}
+
+	return "", false
+}
+
+// Tries to find a Debian copyright file for the package and returns the source URL if IsRepoUrl() agrees.
+func MaybeGetSourceRepoFromDebian(mdir string, pkg string) string {
+	var metadata string
+	if strings.HasPrefix(pkg, "lib") {
+		metadata = path.Join(mdir, "changelogs/main", string(pkg[0:4]), pkg, "unstable_copyright")
+	} else {
+		metadata = path.Join(mdir, "changelogs/main", string(pkg[0]), pkg, "unstable_copyright")
+	}
+	if _, err := os.Stat(metadata); err == nil {
+		// parse the copyright file and go from here
+		Logger.Infof("FYI: Will look at %s", metadata)
+		possibleRepo, ok := MaybeGetSourceFromDebianCopyright(metadata)
+		if !ok {
+			return ""
+		}
+		if IsRepoUrl(possibleRepo) {
+			return possibleRepo
+		}
+		// Incorporate Savannah URL to Git translation here
+		if IsGnuUrl(possibleRepo) {
+			repo, translated := MaybeTranslateSavannahURL(possibleRepo)
+			if translated {
+				return repo
+			}
+		}
+		Logger.Infof("FYI: Disregarding %s", possibleRepo)
+	}
+	return ""
+}
+
+func main() {
+	jsonPath := flag.String("nvd_json", "", "Path to NVD CVE JSON to examine.")
+	debianMetadataPath := flag.String("debian_metadata_path", "", "Path to Debian copyright metadata")
+
+	flag.Parse()
+
+	client, err := logging.NewClient(context.Background(), projectId)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	Logger.GCloudLogger = client.Logger("cpp-osv")
+
+	data, err := ioutil.ReadFile(*jsonPath)
+	if err != nil {
+		Logger.Fatalf("Failed to open file: %v", err) // double check this is best practice output
+	}
+
+	var parsed cves.NVDCVE
+	err = json.Unmarshal(data, &parsed)
+	if err != nil {
+		Logger.Fatalf("Failed to parse NVD CVE JSON: %v", err)
+	}
+
+	for _, cve := range parsed.CVEItems {
+		refs := cve.CVE.References.ReferenceData
+		patch_refs := 0
```

Comment:
nit: fix casing of `patch_refs` to `patchRefs`

Reply:
Done